docs/proposals/20200720-single-controller-multitenancy.md

```diff
@@ -0,0 +1,548 @@
+ clietSecret: wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY
+```
+
+<em>AzureSystemAssignedIdentity</em>
+
+`AzureSystemAssignedIdentity` allows CAPZ to use the SystemAssigned identity provided via Pod 
+Identity.
+
+```yaml
+---
+apiVersion: infrastructure.cluster.x-k8s.io/v1alpha3
+kind: AzureCluster
+metadata:
+  name: "test"
+  namespace: "test"
+spec:
+  region: "westus2"
+  principalRef:
+    kind: AzureSystemAssignedIdentity
+    name: test-account
+---
+apiVersion: infrastructure.cluster.x-k8s.io/v1alpha3
+kind: AzureSystemAssignedIdentity
+metadata:
+  name: "test-account"
+spec:
+  secretRef:
+    name: test-account-creds
+    namespace: capz-system
+  allowedNamespaces:
+  - "test"
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: "test-account-creds"
+  namespace: capz-system
+stringData:
+ tenantID: 6bec3eaa-cac6-11ea-87d0-0242ac130003
+```
+
+Example: AzureUserAssignedIdentity
+
+```yaml
+---
+apiVersion: infrastructure.cluster.x-k8s.io/v1alpha3
+kind: AzureCluster
+metadata:
+  name: "test"
+  namespace: "test"
+spec:
+  region: "westus2"
+  principalRef:
+    kind: AzureUserAssignedIdentity
+    name: test-account
+---
+apiVersion: infrastructure.cluster.x-k8s.io/v1alpha3
+kind: AzureUserAssignedIdentity
+metadata:
+  name: "test-account"
+spec:
+  secretRef:
+    name: test-account-creds
+    namespace: capz-system
+  allowedNamespaces:
+  - "test"
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: "test-account-creds"
+  namespace: capz-system
+stringData:
+ tenantID: 6bec3eaa-cac6-11ea-87d0-0242ac130003
+ clientID: 74e870e6-cac6-11ea-87d0-0242ac130003
+```
+
+### Controller Changes
+
+* If principalRef is specified, the CRD is fetched and unmarshalled into a Azure `Authorizer` for 
+  the principal type.
+* The controller will compare the hash of the credential provider against the same secret’s provider
+  in a cache ([NFR2](#NFR2)).
+* The controller will take the newer of the two and instantiate Azure clients with the selected 
+  `Authorizer`.
+* The controller will set the Identity resource as one of the OwnerReferences of the AzureCluster.
+
+### Clusterctl changes
+
+Today, `clusterctl move` operates by tracking objectreferences within the same namespace, since we 
+are now proposing to use cluster-scoped resources, we will need to add requisite support to 
+clusterctl's object graph to track ownerReferences pointing at cluster-scoped resources, and ensure 
+they are moved. We will naively not delete cluster-scoped resources during a move, as they maybe 
+referenced across namespaces.
+
+#### Validating webhook changes
+
+None.
+
+A validating webhook could potentially handle some cross-resource validation necessary for the [security
+model](#security-model) and provide more immediate feedback to end users. However, it would be 
+imperfect. For example, a change to an Azure Identity resource could affect the validity of 
+corresponding AzureCluster.
+
+#### Principal Type Credential Provider Behaviour
```

Comment:
Conversion changes? What happens if you have a v1alpha3 cluster using SystemAssigned identity and you try to convert to v1alpha2?

Reply:
That's a great question. I think that fails.

We could add the identity information to the metadata of the v1alpha2 version of the AzureCluster. Thoughts?

Reply:
We should at least document that as breaking. Wonder how CAPA is handling conversion... we should probably align on that.